### ISSUE TITLE
Specify the behavior of List

### ISSUE THREAD

**Issue by Xuanwo** [beyondstorage/specs#135]
Here are two problems:

- Service doesn't have a default value for `ListMode`
- Do we need to check `virtual_dir` while `ListMode` is `ModeDir`?

**Comment by JinnyYi** [comment]
> * Service doesn't have a default value for `ListMode`

I wonder whether the default `ListMode` is different for specific service. 
We don't need the `ListMode` parameter like `go-service-fs`. And for object storage like `s3`, the default `ListMode` should be `ListModePrefix`.

**Comment by Xuanwo** [comment]
> I wonder whether the default ListMode is different for specific service.

Yes. The default `ListMode` could be different for different services.

- For fs: could be `ListModeDir` (maybe we can implement `virtual_list_prefix` for fs?)
- Fo3 s3: could be `ListModePrefix`

**Comment by JinnyYi** [comment]
> * For fs: could be `ListModeDir` (maybe we can implement `virtual_list_prefix` for fs?)

I'm not sure the expected operation and results for `virtual_list_prefix`.

**Comment by Xuanwo** [comment]
> I'm not sure the expected operation and results for virtual_list_prefix.

With `virtual_list_prefix`, we will implement prefix listing for fs. (I don't whether we need to do this too)

Maybe we can require all services to implement `ListModeDir` without the check for `virtual_dir`.

**Comment by JinnyYi** [comment]
> With `virtual_list_prefix`, we will implement prefix listing for fs. (I don't whether we need to do this too)

Maybe there's no need to implement `virtual_list_prefix` for fs?
- Seems that we need to list files under the directory first then filter file names according to the prefix. Maybe `ListModeDir` is enough?
- If we support prefix listing for fs, the parameter `path` could not carry enough information. We need to check wether `path` is an existing dir or a dir + prefix?

**Comment by Xuanwo** [comment]
Makes sense. Let's drop out list prefix support for fs until there is a user who requests this feature.
